scripts/tuple_tree_generator/tuple_tree_generator/templates/struct_late.h.tpl

```diff
@@ -46,6 +46,18 @@ template <> struct TupleLikeTraits</*=- struct | user_fullname =*/> {
 };
 
 namespace /*= struct.namespace =*/ {
+
+  /**- if emit_tracking **/
+template <int I> revng::AccessTracker &getTracker(const /*= struct.name =*/ &x) {
+  if constexpr (false)
+    return *std::declval<revng::AccessTracker*>();
+  /**- for field in struct.all_fields **/
+  else if constexpr (I == /*= loop.index0 =*/)
+    return x./*= field.name =*/Tracker;
+  /**- endfor **/
+}
+  /** endif -**/
```

Comment:
If we were to generate a `std::tuple<...>` for both fields and access trackers we could generate less code.
This would just become a `std::get<I>(TrackerTuple)`.

We didn't use `std::tuple` for fields since in the past the fields were public fields, but since we switched to accessors this is no longer required.

What do you think?

Reply:
i did it this way to have simmetry with the get of the field, so you can do get<5> and getTracker<5>, which helps writing easier visits, the tuple idea is viable, but it complicates other generated code because then one has to write get<TTFunctionThatGoesFromFieldToFieldIndex> rather than fieldNameTracker.

no strong opinions toh

Reply:
Next time